vocabularies/Common.json

```diff
@@ -1288,6 +1288,48 @@
       "@Common.Experimental": true,
       "@Core.IsURL": true,
       "@Core.Description": "Base URL for WebSocket connections"
+    },
+    "GetRecommendationsAction": {
+      "$Kind": "Term",
+      "$AppliesTo": ["EntityType"],
+      "@Common.Experimental": true,
+      "@Core.Description": "Name of an action (or function) import that retrieves AI-based recommendations (see [template](#Template_GetRecommendationsAction)) for an entity of the annotated type"
+    },
+    "Template_GetRecommendationsAction": [
+      {
+        "$Kind": "Action",
+        "@Common.Experimental": true,
+        "@Core.Description": "Template for actions that retrieve AI-based recommendations for an entity and its related entities",
+        "@Core.LongDescription": "The template action itself cannot be invoked.",
+        "$Parameter": [
+          {
+            "$Name": "EntityKeys",
+            "$Type": "Edm.ComplexType",
+            "@Core.Description": "Nested structure containing the keys of entities for which recommendations shall be retrieved",
+            "@Core.LongDescription": "The nested complex type of this parameter has the same structure as the entity type bearing the [`GetRecommendationsAction`](#GetRecommendationsAction) annotation\n            and its related entity types, but contains only their key properties.\n            If this parameter can be included in a URL, a function can be used instead of an action."
+          }
+        ],
+        "$ReturnType": { "$Collection": true, "$Type": "Common.RecommendationPropertyType" }
+      }
+    ],
+    "RecommendationPropertyType": {
+      "$Kind": "ComplexType",
+      "@Common.Experimental": true,
+      "target": { "@Core.Description": "Target for this recommendation, like the `target` in [`Messages`](#Messages)" },
+      "proposal": { "$Type": "Common.RecommendationProposalType" }
+    },
+    "RecommendationProposalType": {
+      "$Kind": "ComplexType",
+      "@Common.Experimental": true,
+      "value": {},
+      "text": {},
+      "alternatives": { "$Collection": true, "$Type": "Common.AlternativeRecommendationType" }
+    },
+    "AlternativeRecommendationType": {
+      "$Kind": "ComplexType",
+      "@Common.Experimental": true,
+      "value": {},
```

Comment:
Can we have a description

```json
"value": {
    "@Core.Description": "Recommended value for the property, if it meets a certain criteria."
}
```

Reply:
"If it meets certain criteria" is too vague for a description. What do you want to express?